query-engine/driver-adapters/executor/src/bench.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * Run with: `node --experimental-wasm-modules ./example.js`
+ * on Node.js 18+.
+ */
+import { webcrypto } from "node:crypto";
+import * as qe from "./qe";
+
+import pgDriver from "pg";
+import * as prismaPg from "@prisma/adapter-pg";
+import { DriverAdapter } from "@prisma/driver-adapter-utils";
+
+import { recording } from "./recording";
+import prismaQueries from "../bench/queries.json";
+
+import { run, bench, group, baseline } from "mitata";
+
+import { QueryEngine as WasmBaseline } from "query-engine-wasm-baseline";
+import { QueryEngine as WasmLatest } from "query-engine-wasm-latest";
+
+(global as any).crypto = webcrypto;
+
+async function main(): Promise<void> {
+  // read the prisma schema from stdin
+  const datamodel = await new Promise<string>((resolve, reject) => {
+    let data = "";
+    process.stdin.on("data", (chunk) => {
+      data += chunk;
+    });
+    process.stdin.on("end", () => {
+      resolve(data);
+    });
+    process.stdin.on("error", reject);
+  });
+
+  const url = process.env.DATABASE_URL;
+  if (url == null) {
+    throw new Error("DATABASE_URL is not defined");
+  }
+  const pg = await pgAdapter(url);
+  const { recorder, replayer } = recording(pg);
+
+  await recordQueries(recorder, datamodel, prismaQueries);
+  await benchMarkQueries(replayer, datamodel, prismaQueries);
+}
+
+async function recordQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+): Promise<void> {
+  const qe = await initQeNapiCurrent(adapter, datamodel);
+  await qe.connect("");
+
+  for (const prismaQuery of prismaQueries) {
+    const { description, query } = prismaQuery;
+    console.error("Recording query: " + description);
+    await qe.query(JSON.stringify(query), "", undefined);
+  }
+}
+
+async function benchMarkQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+) {
+  const napi = await initQeNapiCurrent(adapter, datamodel);
+  napi.connect("");
+  const wasmCurrent = await initQeWasmCurrent(adapter, datamodel);
+  wasmCurrent.connect("");
+  const wasmBaseline = await initQeWasmBaseLine(adapter, datamodel);
+  wasmBaseline.connect("");
+  const wasmLatest = await initQeWasmLatest(adapter, datamodel);
+  wasmLatest.connect("");
+
+  try {
+    for (const prismaQuery of prismaQueries) {
+      const { description, query } = prismaQuery;
+      const jsonQuery = JSON.stringify(query);
+
+      group(description, () => {
+        bench("Web Assembly: Baseline", () =>
+          wasmBaseline.query(jsonQuery, "", undefined)
+        );
+
+        bench("Web Assembly: Latest", () =>
+          wasmLatest.query(jsonQuery, "", undefined)
+        );
+
+        baseline("Web Assembly: Current", () =>
```

Comment:
This might be confusing, but `baseline` in here is what do we compare the results to. We want to know how the current WASM engine performs compared to the others. 

"Web Assembly: Baseline" is an old version of the engine against which we can track cumulative improvements.